deps: more permissive mst peerDep, update mst devDep

barbalex opened the pull request:
resolves https://github.com/agilgur5/mst-persist/issues/32

barbalex committed:
Fix upstream dependency conflict

agilgur5 commented on package.json:
```diff
@@ -35,7 +35,7 @@
   },
   "peerDependencies": {
-    "mobx-state-tree": "^3.2.1"
+    "mobx-state-tree": "^5.1.8"
```
So this change would be breaking, instead we can just make the existing range more permissive to allow for any version above:

```suggestion
    "mobx-state-tree": ">=3.2.1"
```

This change would be a patch bump instead

agilgur5 reviewed (changes requested):
You were able to test and confirm that this was working on MST 5, right? Seems like there weren't too many changes in [4.0.0](https://github.com/mobxjs/mobx-state-tree/blob/v5.1.8/changelog.md#400) or [5.0.0](https://github.com/mobxjs/mobx-state-tree/releases/tag/v5.0.0) thankfully

The CI on this repo is still on Travis, so I need to migrate that to GH Actions to get it to run on each PR

agilgur5 reviewed (commented):
We should update the devDeps as well to make sure MST 5 is tested against. That'll require an update of the MobX devDep as well

agilgur5 commented:
> The CI on this repo is still on Travis, so I need to migrate that to GH Actions to get it to run on each PR

CI has been migrated as of #34 now. If you rebase off `main`, that should get it to run on your PR

barbalex commented:
> You were able to test and confirm that this was working on MST 5, right?

Yes, in two of my projects it has been running flawlessly for years

barbalex committed:
make the range more permissive

Co-authored-by: Anton Gilgur <4970083+agilgur5@users.noreply.github.com>

barbalex commented:
Not sure if I did all that's needed. Please check

agilgur5 committed:
deps: update MST + mobx devDeps

- to make sure that we test on MST 5 et al

agilgur5 committed:
ci: run on PR as well

- I thought `push` covered PRs as well, maybe that got changed
  - or it was always only for my own PRs? not sure as it worked in repos I don't own either...

agilgur5 committed:
ci: add skipLibCheck to typecheck

- was getting errors on MST and MobX internal types
  - current TS direction is to use `skipLibCheck` as the default
    - c.f. https://github.com/microsoft/TypeScript-Website/commit/625e1c5f8c4fdf5f5f822e6122ecb9bf2f6e8f33

agilgur5 reviewed (approved):
updated devDeps myself, LGTM now! all tests still passing 🙂 

also had to update some one-line CI settings

agilgur5 commented:
Released in [`0.1.4`](https://github.com/agilgur5/mst-persist/releases/tag/v0.1.4).

Thanks for your continued contribution @barbalex!
